sui_programmability/framework/sources/Balance.move

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// A storable handler for `Coin` balances.
+/// Allows separation of the transferable `Coin` type and the storable
+/// `Balance` eliminating the need to create new IDs for each application
+/// that needs to hold coins.
+module Sui::Balance {
+    friend Sui::Coin;
+
+    const ENONZERO: u64 = 0;
+    const EVALUE: u64 = 0;
+
+    /// Storable balance - an inner struct of a Coin type.
+    /// Can be used to store coins which don't need to have the
+    /// key ability.
+    struct Balance<phantom T> has store {
+        value: u64
+    }
+
+    /// Get the amount stored in a `Balance`.
+    public fun value<T>(self: &Balance<T>): u64 {
+        self.value
+    }
+
+    /// Create an empty `Balance` for type `T`.
+    public fun empty<T>(): Balance<T> {
```

Comment:
This feels a bit asymmetric with `Coin::zero` to me. I don't have a strong preference for `empty` vs `zero`, but I think we should probably pick the same name for both.